### DISCUSSION TITLE
Inconsistency with the omnet++ tictoc tutorial

### DISCUSSION THREAD

**Discussion by anoikob** [omnetpp/omnetpp#1219] [Q&A] [answered]
Greetings. I'm trying to get through the tictoc tutorial, but im stuck on "Turning it Into a Real Network". You see, the tutorial itself has this code and resulting topology:
![image](https://github.com/user-attachments/assets/26bb2276-0fbc-4edc-a0bd-37b519e50efd)

Unfortunately, when I straight up copy-paste it into my omnet++ ned file, I get this:

![image](https://github.com/user-attachments/assets/92f6da99-6460-46ac-938d-0dde6e2dd997)

Any ideas? I'm a complete beginner regarding omnet so I don't know how should I begin fixing this issue. I would also like to note that the .ned file provided within the tutorial ends up producing the same result.

**Comment by rhornig** [comment, marked as answer]
The IDE Ned editor displays the structure of the NED file, i.e. you see NED types in that diagram. Yo see here, that there is a SINGLE vector which is connected to itself (in fact different elements of the vector are connected to each other)

The picture you see in the tutorial is from the runtime environment (Qtenv) (i.e. when you actually run the simulation). In the runtime you see the INSTACES of each vector element so you will see 6 separate instances.

This is normal.

**Reply by anoikob** [reply]
Yes it would seem that you are correct.
So for future reference, when I work with vectors in omnet, I should not trust the .ned editor in the ide, but the simulation gui. Thanks for the answer!

**Reply by rhornig** [reply]
You CAN trust, the NED editor, it's just icons represent different things. In NED editor, they represent the whole vector with displaying the name and the vector size in []. In Qtenv they represent individual modules and [] shows the actual index of the module.

**Reply by anoikob** [reply]
Well yeah, but if I want a specific topology and connections, I won't know if I suceeded in making them unless I open Qtenv

**Reply by rhornig** [reply]
Yes. That's true. As the topology **may** depend on INI files, command line parameters and even the C++ code, the IDE is just trying to estimate the end result (with some serious limitations).
